mono/metadata/class-init.c

```diff
@@ -634,8 +634,9 @@ mono_class_create_from_typedef (MonoImage *image, guint32 type_token, MonoError
 
 	if (tt->rows > tidx){		
 		mono_metadata_decode_row (tt, tidx, cols_next, MONO_TYPEDEF_SIZE);
```

Comment:
You'll also need to check if `cols [MONO_TYPEDEF_FIELD_LIST]` and `cols [MONO_TYPEDEF_METHOD_LIST]` are 0 above as well.  If they are then you'd may end up with a type with no fields having 65535 fields/methods.

Reply:
> You'll also need to check if `cols [MONO_TYPEDEF_FIELD_LIST]` and `cols [MONO_TYPEDEF_METHOD_LIST]` are 0 above as well. If they are then you'd may end up with a type with no fields having 65535 fields/methods.

I wanted to do that, but then there is an explicit check below for `if (cols [MONO_TYPEDEF_FIELD_LIST] &&` https://github.com/Unity-Technologies/mono/pull/2081/files#diff-b5ce0127cae8f7d1681dde065910f7511d50dfff3586d28ec4ea8ba7f37ac256L644 so `first_field_idx` is not used actually

I can refactor the code to make it easier to read, but I didn't want to change too much of the current style 😄

Reply:
yes, if field_count is 0, the index is not used anywhere